core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3189,6 +3301,16 @@ private QueryModel rewriteSelectClause0(final QueryModel model, SqlExecutionCont
                         emitLiterals(qc.getAst(), translatingModel, innerVirtualModel, baseModel, true);
                         continue;
                     } else if (functionParser.getFunctionFactoryCache().isGroupBy(qc.getAst().token)) {
+                        QueryColumn matchingCol = groupByModel.findBottomUpColumnByAst(qc.getAst());
+                        if (useOuterModel && matchingCol != null) {
```

Comment:
nit: would be great to have some comments around this branch.

Reply:
Added in 07bc27b2